File: Algorithms/003.LongestSubstringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length(), i = 0, j = 0, maxLen = 0;
        bool exist[256] = { false };
        while (j < n) {
            if (exist[s[j]]) {
                maxLen = max(maxLen, j-i);
                while (s[i] != s[j]) {
                    exist[s[i]] = false;
                    i++;
                }
                i++;
            } else {
                exist[s[j]] = true;
            }
            j++;
        }
        return max(maxLen, n-i);
    }
};
int main(int argc, char const *argv[])
{
    Solution s;
    cout << s.lengthOfLongestSubstring("wlrbbmqbhcdarzowkkyhiddqscdxrjmowfrxsjybldbefsarcbynecdyggxxpklorellnmpapqfwkhopkmco"
) << endl;
    return 0;
}

File: Algorithms/java/003.LongestSubstringWithoutRepeatingCharacters.java
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length(), i = 0, j = 0, maxLen = 0;
        boolean[] exist= new boolean[256];
        for (int k = 0; k < 256; k++) {
            exist[k] = false;
        }
        while (j < n) {
            if (exist[s.charAt(j)]) {
                maxLen = Math.max(maxLen, j - i);
                while (s.charAt(i) != s.charAt(j)) {
                    exist[s.charAt(i)] = false;
                    i++;
                }
                i++;
            } else {
                exist[s.charAt(j)] = true;
            }
            j++;
        }
        return Math.max(maxLen, n - i);
    }
}
